src/fparser/two/Fortran2003.py

```diff
@@ -3184,6 +3185,138 @@ def match(string):
     match = staticmethod(match)
 
 
+class Cray_Pointer_Stmt(StmtBase, WORDClsBase):  # pylint: disable=invalid-name
+    '''
+    cray-pointer-stmt is POINTER cray-pointer-decl-list
+    '''
+    subclass_names = []
+    use_names = ['Cray_Pointer_Decl_List']
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for a Cray-pointer statement.
+
+        :param string: the reader or string to match as a Cray-pointer \
+        statement.
+        :type string: \
+        :py:class:`fparser.common.readfortran.FortranReaderBase` or \
+        `str`
+        :return: a tuple of size 2 containing a string with the name \
+        "POINTER" and a cray-pointer-decl-list, if there is a match, \
+        or `None` if there is not.
+        :rtype: (str, Cray_Pointer_Decl_List) or None
+
+        '''
+        from fparser.two.utils import EXTENSIONS
+        if not 'cray-pointer' in EXTENSIONS:
+            return None
+        return WORDClsBase.match('POINTER', Cray_Pointer_Decl_List, string,
+                                 require_cls=True)
+
+
+class Cray_Pointer_Decl(Base):  # pylint: disable=invalid-name
+    '''
+    cray-pointer-decl is ( cray-pointer-name, cray-pointee-decl )
+    '''
+    use_names = ['Cray_Pointer_Name', 'Cray_Pointee_Name', 'Cray_Pointee_Decl']
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for a Cray-pointer declaration.
+
+        :param str string: the string to match as a Cray-pointer \
+        declaration.
+        :return: None if there is no match, otherwise a tuple of size \
+        2 containing the name of the pointer as the first argument and \
+        either the name of the pointee as the second argument or a \
+        Cray-pointee declaration.
+        :rtype: None, (Name, Name) or (Name, Cray_Pointee_Decl)
+
+        '''
+        if not string:
+            return None
+        strip_string = string.strip()
+        if not strip_string:
```

Comment:
pylint complains about too many returns. Could remove some by doing e.g.

    if not strip_string or not strip_string.startswith("(") or ...:

Reply:
I could, but I think it is more readable this way. I also play a trump card, which is that we may start returning more detailed syntax errors in the future and separating out the different error conditions allows us to be more specific.

Reply:
OK. Trump card accepted.